packages/vite/src/esbuild-request.ts

```diff
@@ -146,27 +146,33 @@ export class EsBuildModuleRequest implements ModuleRequest {
 
     requestStatus(request.specifier);
 
-    let result = await this.context.resolve(request.specifier, {
-      importer: request.fromFile,
-      resolveDir: dirname(request.fromFile),
-      kind: this.kind,
-      pluginData: {
-        embroider: {
-          enableCustomResolver: false,
-          meta: request.meta,
+    const extensions = ['', '.hbs'];
```

Comment:
@ef4 I tried to move the extension resolving here to defaultResolve, but it does not work.
I think the issue is that when we resolve files in addons, we do use a relative specifier with a from file in the rewritten-addons. What esbuild then sees is something like
```js
{
   type: 'ignored',
   result: {
    external: true,
    path: './components/hello-world'
  }
}
```

so it has relative path as the externalized ID, I think this will confuse the system between esbuild and vite

Reply:
I did the same changes in https://github.com/embroider-build/embroider/pull/1876 and there it works

Reply:
I think we could do this as an intermediate step? I would change the tests to do the checks at runtime